packages/flutter/lib/src/material/tabs.dart

```diff
@@ -660,11 +666,27 @@ class _TabBarState<T> extends ScrollableState<TabBar<T>> implements TabBarSelect
       // want to curve the animation. When _valueIsChanging is false, we're
       // animating based on a pointer event and want linear feedback. It's
       // possible we should move this curve into the selection animation.
-      t = Curves.ease.transform(t);
+      // We animate the leading and trailing edges of the rect differently.
+      // The easiest way to do this is to lerp 2 rects, and piece them together
+      // into 1.
+      Rect leftRect, rightRect;
+      if (_selection.index > _selection.previousIndex) {
+        // Moving to the right - right edge is leading.
+        rightRect = _indicatorTween.lerp(_kTabIndicatorLeadingCurve.transform(t));
+        leftRect = _indicatorTween.lerp(_kTabIndicatorTrailingCurve.transform(t));
+      } else {
+        // Moving to the left - left edge is leading.
+        leftRect = _indicatorTween.lerp(_kTabIndicatorLeadingCurve.transform(t));
+        rightRect = _indicatorTween.lerp(_kTabIndicatorTrailingCurve.transform(t));
+      }
+      _indicatorRect = new Rect.fromLTRB(
+        leftRect.left, leftRect.top, rightRect.right, rightRect.bottom
+      );
+    } else {
+      // TODO(abarth): If we've never gone through handleStatusChange before, we
+      // might not have set up our _indicatorTween yet.
+      _indicatorRect = _indicatorTween.lerp(t);
     }
-    // TODO(abarth): If we've never gone through handleStatusChange before, we
-    // might not have set up our _indicatorTween yet.
-    _indicatorRect = _indicatorTween.lerp(t);
     if (oldRect != _indicatorRect)
       setState(() { });
```

Comment:
while you're here, please put a comment in the block here saying what it is that changed